### ISSUE TITLE
"Wrong hanging indentation" with tabs

### ISSUE THREAD

**Issue by xarx00** [PyCQA/pylint#1148]
### Steps to reproduce
1. Take the following piece code (test.py):

```
class MainFrame(object):
    def init_ui(self):
        self.m_notebook.ChangeSelection(
            self.m_notebook.FindPage(self.m_panelComponents))
```

Note, the indentation is made via tabs, not spaces.
2. `pylint.exe --indent-string='\t' --indent-after-paren=1 --reports=no test.py`
### Current behavior

```
C:  4, 0: Wrong hanging indentation (add 14 spaces).
                        self.m_notebook.FindPage(self.m_panelComponents))
   ^             | (bad-continuation)
```

Adding more tabs or spaces at the beginning of the hanging line doesn't help.
### Expected behavior

I would expect that pylint considers this hanging indentation correct. When the function is placed outside the class, the "Wrong hanging indentation" warning is not reported.
### pylint --version output

```
No config file found, using default configuration
pylint 1.6.4,
astroid 1.4.8
Python 2.7.8 (default, Jul  2 2014, 19:48:49) [MSC v.1500 64 bit (AMD64)]
```

**Comment by moylop260** [comment]
Is duplicated https://github.com/PyCQA/pylint/issues/638?

**Comment by xarx00** [comment]
I don't think it's identical, because
1. in #638, they do not talk about hanging indentation
2. in my case tab indentation seems to be important
But perhaps they are related somehow.

Another example of the same problem (of mine):
```
class FileProcessor(object):
	def read_items(self):
		return [
			XItem(ITEM1),
			XItem(ITEM4),
		]
```
```
C:  4, 0: Wrong hanging indentation (add 14 spaces).
                        XItem(ITEM1),
   ^             | (bad-continuation)
C:  5, 0: Wrong hanging indentation (add 14 spaces).
                        XItem(ITEM4),
   ^             | (bad-continuation)
C:  6, 0: Wrong hanging indentation.
                ]
  ^             || (bad-continuation)
```

**Comment by xarx00** [comment]
I'd say that the problem is caused by incorrect computation of indentation spaces in case of indentation by tabs. 
```
pylint.exe --indent-string='\t' --indent-after-paren=1 --reports=no test.py
```
Pylint wants us to add 14 more spaces (`Wrong hanging indentation (add 14 spaces)`), three characters (tabs) are already present, which gives 17 characters in total. The actual number of spaces seen by pylint is: two tabs before `return` (in the second example), which pylint re-calculates to 16 spaces (2x8), plus 1 space comming from `indent-after-paren`; 17 chars in total again.

When `indent-after-paren` parameter is increased, the number of spaces pylint wants us to add is increased as well.

**Comment by PCManticore** [comment]
Thank you for providing this input, I will take a look when I will get some free time. If you want to try your hand at a patch, please feel free to do so.

**Comment by zacwitte** [comment]
+1
I've been annoyed by this warning as well so turned it off. But hanging indentation is a common problem so I'd like to have this as a part of my linting.

**Comment by dstromberg** [comment]
+1

**Comment by ziesemer** [comment]
+1

**Comment by irridia** [comment]
+1

It would be great to get the patch above applied to mainline...

**Comment by Xcelled** [comment]
+1. @x539, are you going to make a PR?

**Comment by Pierre-Sassoulas** [comment]
`bad-continuation` has been removed in #3571, black or another formatter can help you with this better than Pylint
